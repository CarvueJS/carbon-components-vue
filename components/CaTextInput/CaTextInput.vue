<template>
  <div class="bx--form-item">
    <input id="text-input-1" type="text" class="bx--text-input" :placeholder="placeholder" :value="value" @change="change">
  </div>
</template>

<script>
export default {
  name: 'ca-text-input',
  props: {
    value: null,
    placeholder: {
      type: String,
      default: '',
    },
  },
  methods: {
    change(e) {
      this.$emit('input', e.target.value);
    },
  },
};
</script>

<style></style>
