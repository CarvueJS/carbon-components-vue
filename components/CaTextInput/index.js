import CaTextInput from './CaTextInput';

CaTextInput.install = (Vue) => {
  Vue.component(CaTextInput.name, CaTextInput);
};

export default CaTextInput;
