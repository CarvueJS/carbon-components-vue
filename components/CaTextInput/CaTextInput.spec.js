import { shallow } from 'vue-test-utils';
import CaTextInput from './CaTextInput.vue';

describe('CaTextInput', () => {
  test('is a Vue instance', () => {
    const wrapper = shallow(CaTextInput);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
})
