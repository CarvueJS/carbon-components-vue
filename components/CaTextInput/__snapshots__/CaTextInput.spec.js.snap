// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaTextInput Mathes snapshot 1`] = `
<div class="bx--form-item">
  <input id="text-input-1" type="text" placeholder="" class="bx--text-input">
</div>
`;
